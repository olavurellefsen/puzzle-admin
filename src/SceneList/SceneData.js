export default {
  "data": {
    "scene": [
      {
        "id": 1,
        "title": "køkurin",
        "image_filename": "køkurin.png",
        "initial_puzzle_id": 1,
        "sequence": 1
      },
      {
        "id": 2,
        "title": "sovikamarið",
        "image_filename": "sovikamarið.png",
        "initial_puzzle_id": 10,
        "sequence": 2
      },
      {
        "id": 3,
        "title": "vesið",
        "image_filename": "vesið.png",
        "initial_puzzle_id": 13,
        "sequence": 3
      },
      {
        "id": 4,
        "title": "spælikamarið",
        "image_filename": "spælikamarið.png",
        "initial_puzzle_id": 16,
        "sequence": 4
      },
      {
        "id": 5,
        "title": "gongin",
        "image_filename": "gongin.png",
        "initial_puzzle_id": 19,
        "sequence": 5
      },
      {
        "id": 7,
        "title": "fjósið",
        "image_filename": "fjósið.png",
        "initial_puzzle_id": 25,
        "sequence": 7
      },
      {
        "id": 8,
        "title": "hjallurin",
        "image_filename": "hjallurin.png",
        "initial_puzzle_id": 28,
        "sequence": 8
      }
    ]
  }
}