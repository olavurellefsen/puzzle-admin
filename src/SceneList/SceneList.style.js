import styled from 'styled-components';

export const SceneListContainer = styled.div`
  overflow: auto;  
`;
SceneListContainer.displayName = 'SceneListContainer';