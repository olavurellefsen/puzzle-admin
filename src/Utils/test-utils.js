import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
