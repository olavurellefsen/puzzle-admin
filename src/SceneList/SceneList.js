import React from "react";
import Scene from './Scene/Scene';
import { SceneListContainer } from "./SceneList.style";

const SceneList = () => {
  const scenes = [0,1,2,3];
  return (
    <SceneListContainer>
      {scenes.map((scene, index) => 
        <Scene key={index} sceneId={scene}/>
      )}
    </SceneListContainer>
  );
};
export default SceneList;
