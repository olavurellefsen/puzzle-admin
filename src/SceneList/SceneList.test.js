import React from "react";
import { render, cleanup } from 'test-utils';
import SceneList from "./SceneList";
afterEach(cleanup)

it("should render correctly", () => {
  const { container } = render(<SceneList />);
  expect(container).toMatchSnapshot();
});

it("should show the SceneListContainer element", () => {
  const { queryByTestId } = render(<SceneList />);
  expect(queryByTestId("SceneListContainer")).toBeTruthy();
});