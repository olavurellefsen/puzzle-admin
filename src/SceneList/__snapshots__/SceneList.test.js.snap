// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneList should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SceneList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Scene
          scene={
            Object {
              "id": 1,
              "image_filename": "køkurin.png",
              "initial_puzzle_id": 1,
              "sequence": 1,
              "title": "køkurin",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 2,
              "image_filename": "sovikamarið.png",
              "initial_puzzle_id": 10,
              "sequence": 2,
              "title": "sovikamarið",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 3,
              "image_filename": "vesið.png",
              "initial_puzzle_id": 13,
              "sequence": 3,
              "title": "vesið",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 4,
              "image_filename": "spælikamarið.png",
              "initial_puzzle_id": 16,
              "sequence": 4,
              "title": "spælikamarið",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 5,
              "image_filename": "gongin.png",
              "initial_puzzle_id": 19,
              "sequence": 5,
              "title": "gongin",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 7,
              "image_filename": "fjósið.png",
              "initial_puzzle_id": 25,
              "sequence": 7,
              "title": "fjósið",
            }
          }
        />,
        <Scene
          scene={
            Object {
              "id": 8,
              "image_filename": "hjallurin.png",
              "initial_puzzle_id": 28,
              "sequence": 8,
              "title": "hjallurin",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 1,
            "image_filename": "køkurin.png",
            "initial_puzzle_id": 1,
            "sequence": 1,
            "title": "køkurin",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 2,
            "image_filename": "sovikamarið.png",
            "initial_puzzle_id": 10,
            "sequence": 2,
            "title": "sovikamarið",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 3,
            "image_filename": "vesið.png",
            "initial_puzzle_id": 13,
            "sequence": 3,
            "title": "vesið",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 4,
            "image_filename": "spælikamarið.png",
            "initial_puzzle_id": 16,
            "sequence": 4,
            "title": "spælikamarið",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 5,
            "image_filename": "gongin.png",
            "initial_puzzle_id": 19,
            "sequence": 5,
            "title": "gongin",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 7,
            "image_filename": "fjósið.png",
            "initial_puzzle_id": 25,
            "sequence": 7,
            "title": "fjósið",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "scene": Object {
            "id": 8,
            "image_filename": "hjallurin.png",
            "initial_puzzle_id": 28,
            "sequence": 8,
            "title": "hjallurin",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  overflow: auto;  
",
        ],
      },
      "displayName": "SceneListContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Scene
            scene={
              Object {
                "id": 1,
                "image_filename": "køkurin.png",
                "initial_puzzle_id": 1,
                "sequence": 1,
                "title": "køkurin",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 2,
                "image_filename": "sovikamarið.png",
                "initial_puzzle_id": 10,
                "sequence": 2,
                "title": "sovikamarið",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 3,
                "image_filename": "vesið.png",
                "initial_puzzle_id": 13,
                "sequence": 3,
                "title": "vesið",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 4,
                "image_filename": "spælikamarið.png",
                "initial_puzzle_id": 16,
                "sequence": 4,
                "title": "spælikamarið",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 5,
                "image_filename": "gongin.png",
                "initial_puzzle_id": 19,
                "sequence": 5,
                "title": "gongin",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 7,
                "image_filename": "fjósið.png",
                "initial_puzzle_id": 25,
                "sequence": 7,
                "title": "fjósið",
              }
            }
          />,
          <Scene
            scene={
              Object {
                "id": 8,
                "image_filename": "hjallurin.png",
                "initial_puzzle_id": 28,
                "sequence": 8,
                "title": "hjallurin",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 1,
              "image_filename": "køkurin.png",
              "initial_puzzle_id": 1,
              "sequence": 1,
              "title": "køkurin",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 2,
              "image_filename": "sovikamarið.png",
              "initial_puzzle_id": 10,
              "sequence": 2,
              "title": "sovikamarið",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 3,
              "image_filename": "vesið.png",
              "initial_puzzle_id": 13,
              "sequence": 3,
              "title": "vesið",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 4,
              "image_filename": "spælikamarið.png",
              "initial_puzzle_id": 16,
              "sequence": 4,
              "title": "spælikamarið",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 5,
              "image_filename": "gongin.png",
              "initial_puzzle_id": 19,
              "sequence": 5,
              "title": "gongin",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 7,
              "image_filename": "fjósið.png",
              "initial_puzzle_id": 25,
              "sequence": 7,
              "title": "fjósið",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "scene": Object {
              "id": 8,
              "image_filename": "hjallurin.png",
              "initial_puzzle_id": 28,
              "sequence": 8,
              "title": "hjallurin",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  overflow: auto;  
",
          ],
        },
        "displayName": "SceneListContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
